### PR TITLE
Fix master - actual state...

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

**Comment by sbrunner** [comment]
Needs help to fix the tests...

File: contribs/gmf/test/spec/services/authenticationservice.spec.js [hunk 1]
```diff
@@ -48,7 +48,8 @@ describe('gmf.Authentication', function() {
     gmfAuthentication.load_();
     $httpBackend.flush();
 
-    expect(spy.calls.length).toBe(1);
+    console.log(spy.calls)
```
Comment (fredj) [inline contribs/gmf/test/spec/services/authenticationservice.spec.js]: to be removed

**Comment by sbrunner** [comment]
Just steal one issue with layer opacity ..., does anyone have an idea?

File: contribs/gmf/src/directives/layertree.js [hunk 2]
```diff
@@ -258,11 +258,13 @@ gmf.LayertreeController.prototype.getLayer = function(node, parentCtrl, depth) {
         layer = this.getLayerCaseNotMixedGroup_(node);
         this.prepareLayer_(node, layer);
         break;
-      // no default
+      default:
+        throw 'node wrong type: ' + type;
```
Comment (fredj) [inline contribs/gmf/src/directives/layertree.js]: `goog.asserts.fail(''node wrong type: ' + type')` instead ?

Comment (sbrunner) [inline contribs/gmf/src/directives/layertree.js]: For me it should be an exception, not an assert ...

Comment (gberaudo) [inline contribs/gmf/src/directives/layertree.js]: For exceptions, you should do `throw new Error('node wrong type: ' + type);`.

Comment (sbrunner) [inline contribs/gmf/src/directives/layertree.js]: Fixed, thanks :-)

**Comment by fredj** [comment]
+1

**Pull request merged**